### PR TITLE
Shake to enable/disable

### PR DESCRIPTION
**Pull request by zolomatok**
I ended up swizzling UIApplication to intercept shake events, because extending the ViewController didn't work out.  

For one, I had no luck getting `motionEnded(_:with:)` called on the Debugger's VC. I did manage to get it called on the app's VC, but AFAIK there's some responder chain magic that decides which object gets called with `motionEnded` and which isn't.

Also, detecting shakes on the ViewController seems to require it being the first responder which will not be the case as soon as peeps start editing some text field.

Swizzling UIApplication works great, though.

### CODE REVIEW

**Comment by zolomatok** [comment]
I changed the Build System to `legacy` because of an Xcode / Cocoapods issue regarding the new build system (https://stackoverflow.com/questions/50552752/how-to-rebuild-development-pod-changes)

File: StanwoodDebugger/Modules/Bubble/DebuggerViewController.swift [hunk 1]
```diff
@@ -42,6 +44,21 @@ class DebuggerViewController: UIViewController, DebuggerViewable {
...
+        showShakeHint()
+    }
+    
+    private func showShakeHint() {
+        if !UserDefaults.hintShown {
+            UserDefaults.set(hintShown: true)
+            debuggerButton.shake(toward: .right, amount: 0.18, duration: 1.5, delay: 0.5)
+            DispatchQueue.main.asyncAfter(deadline: .now()+0.8) {
+                var style = ToastStyle(); style.backgroundColor = StanwoodDebugger.Style.tintColor
+                let toast = try! self.view.toastViewForMessage("Shake to disable the Debugger", title: nil, image: nil, style: style)
```
Comment (talezion) [inline StanwoodDebugger/Modules/Bubble/DebuggerViewController.swift]: @zolomatok Let's avoid force try, and change to try?

File: StanwoodDebugger/Modules/Bubble/DebuggerViewController.swift [hunk 2]
```diff
@@ -42,6 +44,21 @@ class DebuggerViewController: UIViewController, DebuggerViewable {
...
         view.addSubview(debuggerButton)
+
+        showShakeHint()
+    }
+    
+    private func showShakeHint() {
+        if !UserDefaults.hintShown {
+            UserDefaults.set(hintShown: true)
+            debuggerButton.shake(toward: .right, amount: 0.18, duration: 1.5, delay: 0.5)
+            DispatchQueue.main.asyncAfter(deadline: .now()+0.8) {
```
Comment (talezion) [inline StanwoodDebugger/Modules/Bubble/DebuggerViewController.swift]: @zolomatok check out Core, we have a nice little main block

File: StanwoodDebugger/Extensions/UserDefaults+Keys.swift [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+//
+
+import Foundation
+
+extension UserDefaults {
+    struct Keys {
+        static let hintShown = "showHint"
+    }
+    
+    static var hintShown: Bool { return UserDefaults.standard.bool(forKey: UserDefaults.Keys.hintShown) }
```
[the remark this answers is not in the document]
Comment (zolomatok) [reply StanwoodDebugger/Extensions/UserDefaults+Keys.swift]: Damn, #function is pure magic. Thanks!
You're right, it's much nicer with `get set`, thanks!

File: StanwoodDebugger/Extensions/UserDefaults+Keys.swift [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+    static func set(hintShown: Bool) { UserDefaults.standard.set(hintShown, forKey: UserDefaults.Keys.hintShown) }
```
[the remark this answers is not in the document]
Comment (zolomatok) [reply StanwoodDebugger/Extensions/UserDefaults+Keys.swift]: You're right, it's much nicer with `get set`, thanks!

File: StanwoodDebugger/Extensions/UserDefaults+Keys.swift [hunk 5]
```diff
@@ -9,9 +9,11 @@ import Foundation
 
 extension UserDefaults {
     struct Keys {
-        static let hintShown = "showHint"
+        static let isHintShown = "isHintShown"
     }
```
Comment (talezion) [inline StanwoodDebugger/Extensions/UserDefaults+Keys.swift]: @zolomatok can be removed now that we use #function

Comment (zolomatok) [reply StanwoodDebugger/Extensions/UserDefaults+Keys.swift]: ohh, right!

**Pull request merged**